### PR TITLE
Update lfe(1) docs

### LINKED ISSUE #286: Document (doc ...)
In recent versions of LFE, and certainly in [1.2.0](https://hex.pm/packages/lfe/1.2.0), we have the built-in shell function, [`doc/1`](https://github.com/rvirding/lfe/blob/1.2.0/src/lfe_shell.erl#L829-L846), but I can't seem to find a description of it in [the documentation](https://github.com/rvirding/lfe/blob/develop/doc/lfe.txt).

### PR DESCRIPTION
**Pull request by yurrriq**
Document `doc/1` and close #286, among other small tweaks. See [commit](https://github.com/rvirding/lfe/commit/dd0ef32af6d060c469aa882f4d32f1b4d3311db5) [messages](https://github.com/rvirding/lfe/commit/b388d883263f172f1336229c7d585950669adeba) for details.

*N.B. While most of the changes pertain only to documentation, I've also snuck in `(h)` as an alias for `(help)`. Let me know if you want me to remove it.*

### CODE REVIEW

**Comment by yurrriq** [comment]
- [x] `s/shell/REPL/` under clear in `doc/lfe.1.md` for continuity's sake

**Comment by rvirding** [comment]
The arguments to doc/describe are wrong, currently they are of the form `mod`, `mod:macro` and `mod:func/arity` for modules, a macro in a module and a function in a module respectively. Though I am open for discussion about what the format should be.

**Comment by yurrriq** [comment]
Ah, good catch. I'll fix it and force push.

File: doc/lfe.txt [hunk 1]
```diff
@@ -22,10 +22,29 @@ BUILT-IN SHELL FUNCTIONS
...
 
+       (clear)
+
+       Clear the REPL output.
+
+       ([doc | describe] Mod)
+
+       ([doc | describe] Mod:Mac)
+
+       ([doc | describe] Mod:Fun/Arity)
```
Comment (yurrriq) [inline doc/lfe.txt:44]: @rvirding, how's that?

**Comment by yurrriq** [comment]
Rebased again.

**Pull request merged**